### PR TITLE
config for the redemption strategy of epsBNBx-BNB collateral

### PR DESCRIPTION
**Pull request by vminkov**
## Description

config for the redemption strategy of epsBNBx-BNB collateral

### CODE REVIEW

**Comment by carlomazzaferro** [comment]
Should we take on already also the redemption strategies needed for the Moonbeam LP tokens?

**Comment by vminkov** [comment]
> While we are at it, should we also take on the redemption strategies needed for the Moonbeam LP tokens?

let's enumerate here those LP tokens and their addresses?

**Comment by carlomazzaferro** [comment]
> > While we are at it, should we also take on the redemption strategies needed for the Moonbeam LP tokens?
> 
> let's enumerate here those LP tokens and their addresses?

Yes, let's do it. will take a stab at it later one

**Comment by carlomazzaferro** [comment]
BSC:

```
 - 0x5c73804FeDd39f3388E03F4aa1fE06a1C0e60c8e (epsBNBx-BNB)
 - 0x80D00D2c8d920a9253c3D65BA901250a55011b37 (mai3EPS)
 - 
 ```

POLYGON:
```
 - 0xb0e69f24982791dd49e316313fD3A791020B8bF7 (WMATIC-MATICx)
 - 0x82d7f08026e21c7713CfAd1071df7C8271B17Eae (MIMO80-PAR20 BLP)
```

MOONBEAM:
```
 - 0xc6e37086D09ec2048F151D11CdB9F9BbbdB7d685 (xcDOT-stDOT)
 - 0x8CCBbcAF58f5422F6efD4034d8E8a3c9120ADf79 (USDC.wh-GLMR)
 - 0xf8f5E8B9Ee84664695B14862212D8092E16331F6 (WBTC.wh-GLMR)
 - 0x8577273FB3B72306F3A59E26ab77116f5D428DAa (WETH.wh-GLMR)
 - 0xa927E1e1E044CA1D9fe1854585003477331fE2Af (DOT.xc-GLMR)
 - 0x61BF1b38930e37850D459f3CB926Cd197F5F88c0 (wstDOT-DOT.xc)
 - 0xB326b5189AA42Acaa3C649B120f084Ed8F4dCaA6 (base4pool)
 - 0x7F5Ac0FC127bcf1eAf54E3cd01b00300a0861a62 (STELLA-GLMR)
```

File: packages/chains/src/polygon/redemptionStrategies.ts [hunk 1]
```diff
@@ -153,6 +153,7 @@ const redemptionStrategies: { [token: string]: [RedemptionStrategyContract, stri
   ],
   [underlying(assets, assetSymbols["MAI-USDC"])]: [RedemptionStrategyContract.UniswapLpTokenLiquidator, USDC],
   [underlying(assets, assetSymbols["WMATIC-MATICx"])]: [RedemptionStrategyContract.UniswapLpTokenLiquidator, WMATIC],
+  [underlying(assets, assetSymbols.MIMO_PAR_80_20)]: [RedemptionStrategyContract.BalancerLpTokenLiquidator, "1"],
```
Comment (vminkov) [inline packages/chains/src/polygon/redemptionStrategies.ts:156]: this should not be `"1"` - it should be the the asset symbol of the output token - e.g. `assetSymbols.PAR`

Comment (carlomazzaferro) [reply packages/chains/src/polygon/redemptionStrategies.ts:156]: Actually it should be an underlying address, not a symbol

Comment (carlomazzaferro) [reply packages/chains/src/polygon/redemptionStrategies.ts]: @vminkov review here: using PAR as output, how would we get USDC back instead?

File: packages/sdk/src/modules/liquidation/redemptionStrategy.ts [hunk 2]
```diff
@@ -115,9 +113,34 @@ const getStrategyAndData = async (fuse: MidasBase, inputToken: string): Promise<
...
+      if (
+        preferredOutputToken == ethers.constants.AddressZero ||
+        preferredOutputToken == "0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE"
+      ) {
+        actualOutputToken = fuse.chainSpecificAddresses.W_TOKEN;
+      }
+      return {
+        strategyAddress: redemptionStrategyContract.address,
+        strategyData: new ethers.utils.AbiCoder().encode(
+          // TODO: double check
```
Comment (carlomazzaferro) [inline packages/sdk/src/modules/liquidation/redemptionStrategy.ts:139]: @vminkov double check here

File: packages/sdk/src/modules/liquidation/redemptionStrategy.ts [hunk 3]
```diff
@@ -115,9 +113,34 @@ const getStrategyAndData = async (fuse: MidasBase, inputToken: string): Promise<
...
+        outputToken: actualOutputToken,
+      };
+    case RedemptionStrategyContract.SaddleLpTokenLiquidator:
+      const saddleLpOracleAddress = fuse.chainDeployment.CurveLpTokenPriceOracleNoRegistry.address;
+      const saddleLpOracle = new Contract(saddleLpOracleAddress, SaddleLpPriceOracleAbi, fuse.provider);
+
+      tokens = await saddleLpOracle.callStatic.getUnderlyingTokens(inputToken);
+      preferredOutputToken = pickPreferredToken(fuse, tokens, outputToken);
+
+      // the native asset is not a real erc20 token contract, converting to wrapped
```
Comment (carlomazzaferro) [inline packages/sdk/src/modules/liquidation/redemptionStrategy.ts:128]: @vminkov double check if these checks are needed here as well as in contracts

File: packages/sdk/src/modules/liquidation/redemptionStrategy.ts [hunk 4]
```diff
@@ -208,24 +231,26 @@ const getStrategyAndData = async (fuse: MidasBase, inputToken: string): Promise<
 
       return { strategyAddress: redemptionStrategyContract.address, strategyData, outputToken };
     }
-    default: {
-      return { strategyAddress: redemptionStrategyContract.address, strategyData: [], outputToken };
-    }
-  }
-};
+    case RedemptionStrategyContract.BalancerLpTokenLiquidator: {
+      // TODO: do we need this at all? Or can we just use the output token directly?
```
Comment (carlomazzaferro) [inline packages/sdk/src/modules/liquidation/redemptionStrategy.ts:235]: @vminkov double check if we need this logic at all, or if we rely on the `outputToken` that is set in the `redemptionStrategies.ts`

File: packages/chains/src/polygon/redemptionStrategies.ts [hunk 5]
```diff
@@ -152,6 +152,8 @@ const redemptionStrategies: { [token: string]: [RedemptionStrategyContract, stri
     underlying(assets, assetSymbols.USDC),
   ],
   [underlying(assets, assetSymbols["MAI-USDC"])]: [RedemptionStrategyContract.UniswapLpTokenLiquidator, USDC],
+  [underlying(assets, assetSymbols["WMATIC-MATICx"])]: [RedemptionStrategyContract.UniswapLpTokenLiquidator, WMATIC],
+  [underlying(assets, assetSymbols.MIMO_PAR_80_20)]: [RedemptionStrategyContract.BalancerLpTokenLiquidator, PAR],
```
Comment (carlomazzaferro) [inline packages/chains/src/polygon/redemptionStrategies.ts:156]: @vminkov review here: using PAR as output, how would we get USDC back instead?

Comment (vminkov) [reply packages/chains/src/polygon/redemptionStrategies.ts:156]: We need a redemption strategy for that, check the on-chain liquidity for such sources

Comment (carlomazzaferro) [reply packages/chains/src/polygon/redemptionStrategies.ts:156]: Sadly there's not much, except.... 

jEUR-PAR curve: https://polygon.curve.fi/factory/285

Then redeem jEUR for USDC. Wdyt?

Comment (carlomazzaferro) [reply packages/chains/src/polygon/redemptionStrategies.ts]: https://github.com/Midas-Protocol/monorepo/pull/1185/commits/7f922552b60c8c0f15447a2395b43738edd35074

Not entirely sure this is correct

File: packages/chains/src/arbitrum/assets.ts [hunk 6]
```diff
@@ -122,6 +125,32 @@ export const assets: SupportedAsset[] = [
...
+  {
+    symbol: assetSymbols.USDs,
+    underlying: USDs,
+    name: "Sperax USD",
+    decimals: 18,
+    oracle: OracleTypes.UniswapV3PriceOracle,
+    extraDocs: defaultDocs("https://arbiscan.io", USDs),
+    disabled: true,
+  },
+  // Awaiting price feeds
```
Comment (vminkov) [inline packages/chains/src/arbitrum/assets.ts:137]: what price feeds are awaiting here if they are uni v3 and can use the embedded price feed?

**Review by carlomazzaferro — APPROVED** [review]

**Pull request merged**